cypress/support/commands.js

```diff
@@ -737,66 +737,96 @@ Cypress.Commands.add('uploadSubmissionFiles', (files, options) => {
 		return;
 	}
 
-	options = {
-		uploadUrl: /submissions\/\d+\/files$/,
-		editUrl: /submissions\/\d+\/files\/\d+/,
-		primaryFileGenres: ['Article Text', 'Book Manuscript', 'Chapter Manuscript'],
-		...options
-	};
-
-	// Setup upload listeners
-	cy.server();
-
-	cy.route({
-		method: "POST",
-		url: options.uploadUrl,
-	}).as('fileUploaded');
-
-	cy.route({
-		method: "POST",
-		url: options.editUrl
-	}).as('genreDefined');
-
-	files.forEach(file => {
-		cy.fixture(file.file, 'base64').then(fileContent => {
-
-			// Upload the file
-			cy.get('input[type=file]').attachFile(
-				{
-					fileContent,
-					encoding: 'base64',
-					filePath: file.fileName,
-					mimeType: file.mimetype,
-				}
-			);
-			cy.wait('@fileUploaded').its('status').should('eq', 200);
-
-			// Set the file genre
-			const $row = cy.get('a:contains("' + file.fileName + '")').parents('.listPanel__item');
-			$row.contains('What kind of file is this?');
-			if (options.primaryFileGenres.includes(file.genre)) {
-				// For some reason this is locating two references to the button,
-				// so just click the last one, which should be the most recently
-				// uploaded file.
-				$row.get('button:contains("' + file.genre + '")').last().click();
-			} else {
-				$row.get('button:contains("Other")').last().click();
-				cy.get('.pkpFormField--options__optionLabel').contains(file.genre).click();
-				cy.get('.modal button').contains('Save').click();
+	if (Cypress.env('contextTitles').en_US == 'Public Knowledge Preprint Server') {
```

Comment:
Rather than backing this into the `uploadSubmissionFiles` command, why not write a separate command, `uploadGalleyFiles` or `createGalleys`? This command could be re-used (for example, in OJS it could be used to add galleys from the publication page).

Then in OPS you don't need to call `uploadSubmissionFiles` at all.

Reply:
I've given this a crack; see: https://github.com/pkp/pkp-lib/issues/7191#issuecomment-1320752597